src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -76,11 +76,18 @@ internal class LaunchSettingsProvider : OnceInitializedOnceDisposed, ILaunchSett
                                                                                                                     project);
             SourceControlIntegrations = new OrderPrecedenceImportCollection<ISourceCodeControlIntegration>(projectCapabilityCheckProvider: project);
 
+            DefaultLaunchProfileProviders = new OrderPrecedenceImportCollection<IDefaultLaunchProfileProvider>(ImportOrderPrecedenceComparer.PreferenceOrder.PreferredComesFirst, project);
+
             _projectSubscriptionService = projectSubscriptionService;
             _appDesignerSpecialFileProvider = appDesignerSpecialFileProvider;
             _projectFaultHandler = projectFaultHandler;
             _launchSettingsFilePath = new AsyncLazy<string>(GetLaunchSettingsFilePathNoCacheAsync, commonProjectServices.ThreadingService.JoinableTaskFactory);
 
+            _defaultLaunchProfile = new Lazy<ILaunchProfile?> (() => {
+                    return DefaultLaunchProfileProviders?.First()?.Value?.CreateDefaultProfile();
```

Comment:
You need `FirstOrNull` as `First` will throw if the collection is empty.

Reply:
I fixed the others (will send the commit in a minute). On this one, though, if no defaultlaunchprofileprovider exists, MEF will fail the import. We'll never get this far. I'm not checking if _defaultLaunchProfile is null when I use it for this reason. That's also what's failing the test right now. If you want I'll do FirstOrDefault, no problem... but I think First makes the fact that null is not an option very explicit.

Reply:
The import shouldn't fail, even if there are no `IDefaultLaunchProfileProvider`s. The `ProjectSystemContract` attribute explicitly states that there can be zero or more implementations.

Reply:
To fix the test I think you can just update `LaunchSettingsProviderTests.GetLaunchSettingsProvider` to explicitly add an instance of the `DefaultLaunchProfileProvider` to the created `LaunchSettingsUnderTest` object.

Reply:
Thanks Tom, I'll correct those two things and ping you again